scripts/test-cache.sh

```
@@ -0,0 +1,487 @@
+    fi
+}
+
+# Test execution with timing
+run_cache_test() {
+    local test_name="$1"
+    local url="$2"
+    local expected_status="${3:-200}"
+
+    TOTAL_TESTS=$((TOTAL_TESTS + 1))
+
+    local start_time=$(python3 -c "import time; print(int(time.time() * 1000))" 2>/dev/null || echo "0")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)